signing/src/main/java/tech/pegasys/web3signer/signing/config/DefaultArtifactSignerProvider.java

```diff
@@ -60,6 +71,49 @@ public Future<Void> load() {
                       }))
               .forEach(signers::putIfAbsent);
 
+          // for each loaded signer, load commit boost proxy signers (if any)
+          commitBoostKeystoresParameters.ifPresent(
+              keystoreParameter -> {
+                if (!keystoreParameter.isEnabled()) {
+                  return;
+                }
+                signers.forEach(
+                    (identifier, __) -> {
+                      final Path identifierPath =
+                          keystoreParameter.getKeystoresPath().resolve(identifier);
+                      if (identifierPath.toFile().canRead()
+                          && identifierPath.toFile().isDirectory()) {
+                        final Path v4Dir = identifierPath.resolve("v4");
+
+                        if (v4Dir.toFile().canRead() && v4Dir.toFile().isDirectory()) {
```

Comment:
Why are separate directories for each proxy key version needed? Pretty sure the file formats have a version

Reply:
This allow easier loading using existing BLS and SECP bulkloading utility classes. Otherwise, I would have to parse json, check the version, and then invoke BLS or SECP decryption logic. Since the proxy keys Generate logic (the other PR) is controlled programmatically, the user is not involved in placing these files.

Reply:
each delegator (BLS Signer) can have BLS Proxy Signers and SECP Proxy Signers. Hence:
```
/commit_boost_keystores_dir
..../0xblskey0..n/
.........../0xblskey0..n/v3/0xproxy_secp_key0...n
.........../0xblskey0..n/v4/0xproxy_bls_key0...n

```

Reply:
as discussed, updated to use BLS and SECP256K1 as subdirectories for each signer public key.